File: src/01-easy/lumen/solution.scala
import scala.io.StdIn
object Solution{def main(a:Array[String]):Unit={
    val n=StdIn.readInt();val l=StdIn.readInt()
    val g=Array.fill(n)(StdIn.readLine().trim.split(" "))
    var d=0
    for(r<-0 until n;c<-0 until n){
        val lit=(0 until n).exists(r2=>(0 until n).exists(c2=>g(r2)(c2)=="C"&&math.max(math.abs(r-r2),math.abs(c-c2))<l))
        if(!lit) d+=1
    }
    println(d)
}}

File: src/01-easy/lumen/solution.pl
$| = 1;
my $n = <STDIN>+0; my $l = <STDIN>+0;
my @g; for (0..$n-1) { push @g, [split / /, <STDIN>]; }
my $d = 0;
for my $r (0..$n-1) { for my $c (0..$n-1) { my $lit=0;
    for my $r2 (0..$n-1) { for my $c2 (0..$n-1) {
        my $dr=abs($r-$r2); my $dc=abs($c-$c2); my $mx=$dr>$dc?$dr:$dc;
        $lit=1 if $g[$r2][$c2] eq 'C' && $mx < $l;
    }}
    $d++ unless $lit;
}}
print "$d\n";
